Feature/amazon q /doc: upload diagram files during doc generation

vikshe opened the pull request:
Upload architecture diagram image files during documentation generation

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X ] New feature (non-breaking change which adds functionality)

## Description
Upload architecture diagram image files during documentation generation

## Checklist
- [ X ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ X] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

avi-alpert commented on plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/FeatureDevSessionContext.kt:
```diff
@@ -43,9 +43,17 @@ interface RepoSizeError {
     // TODO: Need to correct this class location in the modules going further to support both amazonq and codescan.
 
+    // well known source files that do not have extensions
+    private val wellKnownSourceFiles = setOf(
```
this change just moves wellKnownSourceFiles further up right? just wanted to make sure there's no actual change here to /dev's functionality

vikshe replied:
for this variable - yes, only move it up. 

the whole file has same functionality as before,  the only changes is to add support to extend this class so that our class can include diagrams in the zip archive

linyuxi-0511 committed:
fix(amazonq): filter out frequent client error from codegeneration telemetry and ad… (#5345)

* Filter out frequent client error from codegeneration telemetry and add errMsg to reasonDesc when desc is null

* refactor error messages into constnt

---------

Co-authored-by: Yuxin Lin <linyuxi@amazon.com>

chungjac committed:
fix(amazonq): enable syntax highlighting when viewing diff for generated unit tests (#5337)

* enable syntax highlighting when viewing diff for generated unit tests

changelog

remove unused import

* lint

* virtualFile always created and will not be null

* use session.testFileName instead of extension

* lint

* create temporary virtual file

* comment

* Update bugfix-3b7843b4-9673-4afa-90e8-7c4278f0d0fe.json

laileni-aws committed:
fix(amazonq): Truncating user prompt for unit test generation. (#5359)

* Adding truncation logic to testGen user prompt

KevinDing1 committed:
telemetry(amazonq): refactor doc telemetry to v2 version (#5316)

laileni-aws committed:
fix(amazonq): Q infers source file from active test file for UTG. (#5356)

* Show the source file path instead of test file path in short answer if user does /test on test file.

LokeshDogga13 committed:
Updated AuthTelemetry.addConnection telemetry event to Telemetry.auth.addConnection (#5338)

* Updated AddConnection telemetry event in LoginBrowser.kt

* Updated AddConnection telemetry event

* Fixed issues using gradlew detekt

dhasani23 committed:
Revert "revert(amazonq): hide java 21 option (#5354)" (#5358)

This reverts commit 9a60e50b02f9941f3ea4bbe2893f43fd27ba2346.

rli committed:
build: fix 251 build failure from resolving ui-tests-starter dependencies (#5329)

ui-tests-starter attempts to resolve the IDE distribution and fails, but it should not actually be required to build/execute

rli committed:
build: include 2025.1 in prerelease automation (#5330)

ashishrp-aws committed:
Adding gitignore usecase for Test Generation to add more extensions and folders to ignore list. (#5361)

evanliu048 committed:
config(amazonq):  customization override should respect users' selection (#5347)

config(amazonq): customization override should respect users' selection 

---------

Co-authored-by: Richard Li <742829+rli@users.noreply.github.com>

ashishrp-aws committed:
fix(amazonq): fix for /test authentication check (#5355)

* Correcting auth checks for /test workflow

zuoyaofu committed:
fix(amazonq): code review navigation not focusing selected issues (#5326)

* fix(amazonq): single click or select through arrow keys on review code issues does not focus code editor to line nor show issue details

* add change report

* add back modelzonemarket.cs file

* deleted irrelevant comment

---------

Co-authored-by: Tai Lai <tlaiamz@amazon.com>

evanliu048 committed:
Revert "feat(amazonq): Prefetch next inline recommendation (#5290)" (#5368)

This reverts commit 805cb399c5baf865bc811ed76c5ee5ed943a5a62.

rli committed:
ci: use single matrix for all prerelease artifacts (#5362)

It is easy to forget to update all artifacts on a new MV, so consolidate all artifacts under same matrix

rli committed:
ci: steal commit message linter from VSC toolkit (#5351)

Attempt to enforce more consistency on commit messages

Commit:
Updating version to 3.55

Commit:
Updating SNAPSHOT version to 3.56-SNAPSHOT

leigaol committed:
refactor(amazonq): improve error handling of workspace index (#5373)

* improve error handling of workspace index

Will-ShaoHua committed:
refactor(Q tech debt): make cwsprClientAdapr stateless and obtain sdk client from AwsClientManager (#5331)

* No need to store a client as a state within this class, AwsClientManager already handle the connection change

resolving comments #5290 (comment)

dhasani23 committed:
fix(amazonq): only search for sql strings in java files (#5376)

Co-authored-by: David Hasani <davhasan@amazon.com>

rli committed:
fix(telemetry): fix region warning when initializing telemetry with no user-selected region (#5365)

WARN - software.aws.toolkits.jetbrains.core.credentials.AwsConnectionManager - Using activeRegion when region is null, calling code needs to be migrated to handle null
java.lang.IllegalStateException
	at software.aws.toolkits.jetbrains.core.credentials.AwsConnectionManager.getActiveRegion(AwsConnectionManager.kt:224)
	at software.aws.toolkits.jetbrains.core.credentials.AwsConnectionManagerKt.activeRegion(AwsConnectionManager.kt:370)
	at migration.software.aws.toolkits.jetbrains.services.telemetry.TelemetryService.record(TelemetryService.kt:59)
	at software.aws.toolkits.telemetry.SessionTelemetry.start(SessionTelemetry.kt:287)
	at software.aws.toolkits.telemetry.SessionTelemetry.start$default(SessionTelemetry.kt:271)
	at software.aws.toolkits.jetbrains.services.telemetry.AwsToolkitStartupMetrics.execute$lambda$0(AwsToolkitStartupMetrics.kt:14)
	at com.intellij.ide.util.RunOnceUtilKt.doRunOnce(RunOnceUtil.kt:68)
	at com.intellij.ide.util.RunOnceUtilKt.access$doRunOnce(RunOnceUtil.kt:1)

rli committed:
fix: Fix idea.log path referenced in plugin run configurations (#5369)

Since 2.0, logs are isolated by platform/version pair

Will-ShaoHua committed:
revert(amazonq): pr #5331 broke tests by throwing NotAMock (#5383)

This reverts commit 49719934bd36af014983dc91a0c34528e16cea0f.

Randall-Jiang committed:
telemetry(amazonq) add feedback functionality with thumbs up/down (#5336)

Previously, UTG did not have telemetry to collect user feedback on generated unit tests. This PR introduces telemetry support to address that gap. Additionally, it enables users to submit comments, which will be routed to a private SIM folder.

rli committed:
build(243): fix build failures from rider failing to resolve test framework (#5385)

Force resolve 243.21565.192 -> 243.21565.193
```
org.gradle.internal.resolve.ModuleVersionNotFoundException: Could not find any version that matches com.jetbrains.intellij.java:java-test-framework:{strictly [243, 243.21565.192]; prefer 243.21565.192}.
Versions that do not match:
  - 243.24978.46
  - 243.23654.189
  - 243.23654.153
  - 243.23654.117
  - 243.22562.218
  - + 124 more
```

laileni-aws committed:
telemetry(amazonq): Aligning telemetry strings in VSC and JB. (#5382)

* Adding truncation logic to testGen user prompt

* Adding change log

* Updating telemetry error message to align VSC and JB

* Removing unwanted files

gandhi-21 committed:
fix(amazonq): update logic for converting gitignore pattern to regex (#5360)

* fix(amazonq): update logic for converting gitignore pattern to regex

* fix(amazonq): update logic for converting gitignore pattern to regex

---------

Co-authored-by: Gaurav Gandhi <gangaur@amazon.com>
Co-authored-by: Richard Li <742829+rli@users.noreply.github.com>

Will-ShaoHua committed:
log(amazonq): update log string when workspace process is not spinned up (#5375)

andrewyuq committed:
fix(amazonq): fix tab acceptAction not working in JB remote (#5379)

* fix tab acceptAction not working in JB remote

* use isRunningOnRemoteBackend

vikshe committed:
fix(amazonq): /doc add suppoort for uploading architecture diagrams (#5357)

Co-authored-by: Viktor Shesternyak <vshester@amazon.com>

ashishrp-aws committed:
Adding more extensions and no extension files to test generation usecase ignore list. (#5389)

ctlai95 committed:
config(amazonq): update polling config for codefix (#5390)
